src/mongocxx/private/collection.hpp

```diff
@@ -39,6 +39,13 @@ class collection::impl {
           client_impl(client) {
     }
 
+    impl(const impl& i) : collection_t{libmongoc::collection_copy(i.collection_t)} { }
```

Comment:
That function scares me a bit. It is global, and needs to be done during process setup. I'm not sure I want to mess with it. My understanding is that libbson / libmongoc already assume a non-failing allocator, and they can't throw. I don't think it helps us